### PR TITLE
Update enum types

### PR DESCRIPTION
**Pull request by komalali**
This PR updates the enum types to use the new generated types and use them as relaxed enums with a fallback to string.

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -561,7 +561,7 @@ func Provider() tfbridge.ProviderInfo {
 					},
 					"metrics_granularity": {
 						Type:     "string",
-						AltTypes: []tokens.Type{awsType(autoscalingMod, "metrics", "MetricsGranularity")},
+						AltTypes: []tokens.Type{awsType(autoscalingMod, "MetricsGranularity", "MetricsGranularity")},
```
Comment (stack72) [inline provider/resources.go:564]: How is this manifested to the end user (if at all)?

**Review by stack72 — APPROVED** [review]
This is great - would it be possible to add a test that uses an enum?

File: provider/resources.go (hunk 1 above)
Comment (komalali) [reply provider/resources.go:564]: It doesn't result in any impactful user-facing changes, just this: https://github.com/pulumi/pulumi-aws/blob/master/examples/lambda-layer-new/index.ts#L17-L18

**Comment by komalali** [comment]
@stack72 There are a [couple](https://github.com/pulumi/pulumi-aws/blob/master/examples/lambda-layer-new/index.ts#L17-L18) [examples](https://github.com/pulumi/pulumi-aws/blob/master/examples/serverless-raw/index.ts#L44) that use the TS enums, but I'm going to make another PR with examples updated to use it in each language.

**Comment by stack72** [comment]
Yeah this was just a follow up not a blocker :) merge away!

**Pull request merged**